testing/endpoint.go

```diff
@@ -323,8 +328,22 @@ func (endpoint *Endpoint) QueryConnectionHandshakeProof() (
 	return clientState, clientProof, consensusProof, consensusHeight, connectionProof, proofHeight
 }
 
+var sequenceNumber int
+
+// IncrementNextChannelSequence incrementes the value "nextChannelSequence" in the store,
+// which is used to determine the next channel ID.
+// This guarantees that we'll have always different IDs while running tests.
+func (endpoint *Endpoint) IncrementNextChannelSequence() {
+	if endpoint.disableUniqueChannelIDs {
+		return
+	}
+	sequenceNumber++
```

Comment:
We actually thought of a few different approaches like this. Initializing B at some higher value, maintaining some mapping on the endpoints or paths (since channel IDs need to just be unique between chains on a single path within the test) etc.

For example something like this

```golang
func NewPath(chainA, chainB *TestChain) *Path {
	endpointA := NewDefaultEndpoint(chainA)
	endpointB := NewDefaultEndpoint(chainB)

	// initialize chainB with a higher sequence number to avoid duplicate channel IDs within tests.
	chainB.App.GetIBCKeeper().ChannelKeeper.SetNextChannelSequence(chainB.GetContext(), uint64(1000))

	endpointA.Counterparty = endpointB
	endpointB.Counterparty = endpointA

	return &Path{
		EndpointA: endpointA,
		EndpointB: endpointB,
	}
}

```

But this ends up failing on this test https://github.com/cosmos/ibc-go/blob/b61a35a4d25f6c71983db814ab03ab8f91efc349/modules/core/keeper/msg_server_test.go#L1587-L1624

because of the re-use of channel-1000

(a bunch of tests fail, this is just used as an example)

This test passes if we go back to using a global to track number of paths created

```golang
const channelOffset = 100

var createdPaths = 0

// NewPath constructs an endpoint for each chain using the default values
// for the endpoints. Each endpoint is updated to have a pointer to the
// counterparty endpoint.
func NewPath(chainA, chainB *TestChain) *Path {
	createdPaths++

	endpointA := NewDefaultEndpoint(chainA)
	endpointB := NewDefaultEndpoint(chainB)

	// initialize chainB with a higher sequence number to avoid duplicate channel IDs within tests.
	chainB.App.GetIBCKeeper().ChannelKeeper.SetNextChannelSequence(chainB.GetContext(), uint64(createdPaths*channelOffset))

	endpointA.Counterparty = endpointB
	endpointB.Counterparty = endpointA

	return &Path{
		EndpointA: endpointA,
		EndpointB: endpointB,
	}
}
```

This one actually does pass all but the reworked happy path case of `TestQueryIncentivizedPacketsForChannel`  (and the ModelBasedRelay tests )

I think we can remove the potential issues of this number going out of sync if we modify it here right? 

We still need the package level variables AFAIK unless we want to re-work effected tests.

I think a package level variable here that is used in NewPath is safer than one used the way it is now in the endpoint helper fns.

WDYT @colin-axner  / @bznein